### PR TITLE
WIP: Add doc for CVE-2017-14491

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

File: docs/advisories/cve_2017_14491.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# CVE-2017-14491
+
+A vulnerability in dnsmasq, used by kube-dns, requires an upgrade to the kube-dns component.
+
+[More details](https://security.googleblog.com/2017/10/behind-masq-yet-more-dns-and-dhcp.html)
+
+## Fixed in kops versions:
+
+Not yet released
```
Comment (mikesplain) [inline docs/advisories/cve_2017_14491.md]: May want to link to the PR for future context: https://github.com/kubernetes/kops/pull/3511

**Comment by justinsb** [comment]
Sorry @mikesplain  - I just revised it to the minimal fix and force-pushed - I am a bad person.

I will link the PR - good suggestion!

**Comment by chrislovecnm** [comment]
Can we add instructions for 1.5 - kubectl set image deployment/kube-dns dnsmasq=gcr.io/google_containers/k8s-dns-dnsmasq-amd64:1.14.5 --namespace=kube-system

I need to validate these

**Comment by justinsb** [comment]
Note - these instructions don't work for all versions.  If you have applied them, please double-check that your kube-dns pods are healthy.

I made a mistake on one of my clusters and conflated the 1.5 and the 1.6/1.7 instructions.  If you apply the change, make double sure that `kubectl get pods -n kube-system | grep kube-dns` is showing all the containers running, and not 2/3 or 3/4

File: docs/advisories/cve_2017_14491.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+## Fixed in kops versions:
+
+Not yet released.  Fixed by @mikesplain in [#3511](https://github.com/kubernetes/kops/pull/3511)
+
+## Hotfix for Kubernetes
+
+The minimal fix is to just update the image for the dnsmasq pod:
+
+```
+kubectl set image deployment/kube-dns -n kube-system dnsmasq=gcr.io/google_containers/k8s-dns-dnsmasq-amd64:1.14.5
```
Comment (georgebuckerfield) [inline docs/advisories/cve_2017_14491.md]: Should the image name here be `gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.5`?

**Comment by georgebuckerfield** [comment]
Should the image name in the hotfix be `gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.5`?

**Comment by chrislovecnm** [comment]
@justinsb we had a user run into the missing configmap problem btw. 
 The issue is updated https://github.com/kubernetes/kops/issues/3512 - how do we want to handle the instructions for 1.4 and 1.5, include a note about adding the configmap if it is missing?

**Comment by chrislovecnm** [comment]
Closing because I pr'ee the doc per Justin

**Pull request closed without merging**